### PR TITLE
DDF-04275 Replace keyczar with tink

### PR DESCRIPTION
**Pull request by bakejeyner**
### NOTE: This is a cherry pick from https://github.com/codice/ddf/pull/4339. The following files are the only files which required resolving during the cherry pick:
```
pom.xml
distribution\docs\src\main\resources\content\_dependencyList\ddf-dependency-list.adoc
distribution\docs\src\main\resources\content\_securityFramework\security-encryption-intro.adoc
```

Uncomment this block and include PR # of change against 2.13.x where appropriate.
When doing so, it is reasonable to delete much of the rest of the PR description, leaving only:
- This uncommented block with link to 2.13.x PR
- Reviewers tagged
- Teams tagged
- Committers tagged
- The link to the "Notes on Review Process"

##### ABBREVIATED REVIEW BETWEEN 2.13.X AND MASTER IS IN EFFECT
Link to 2.13.x PR: https://github.com/codice/ddf/pull/4339

#### What does this PR do?
Replaces keyczar with tink.

#### Who is reviewing it? 
@blen-desta 
@brjeter 
@coyotesqrl 
@garrettfreibott 
@stustison 

#### How should this be tested?
See test steps on 2.13.x PR.

#### What are the relevant tickets?
Fixes: #4275

#### Checklist:
- [x] Documentation Updated
- [ ] Update / Add Threat Dragon models
- [x] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

**Comment by bakejeyner** [comment]
build now

**Comment by bakejeyner** [comment]
build now

File: platform/security/encryption/platform-security-encryption-impl/pom.xml [hunk 1]
```diff
@@ -55,12 +60,57 @@
                     <instructions>
                         <Bundle-SymbolicName>${project.artifactId}</Bundle-SymbolicName>
                         <Embed-Dependency>
-                            keyczar,
```
Comment (brjeter) [inline platform/security/encryption/platform-security-encryption-impl/pom.xml]: ❗️ It looks like` keyczar` isn't being removed from the pom here.

File: pom.xml [hunk 2]
```diff
@@ -314,6 +314,8 @@
         <failsafe.argline>${jacoco.argline}</failsafe.argline>
 
         <fabric8.docker.plugin.version>0.27.1</fabric8.docker.plugin.version>
+        <tink.version>1.2.2</tink.version>
+        <org.json.version>20170516</org.json.version>
```
Comment (brjeter) [inline pom.xml]: ❓ We updated the `protobuf.version` to `3.6.1` on the release branch. Should we not also update it from `3.5.1` to `3.6.1` here on `master`?

Comment (bakejeyner) [reply pom.xml]: Great catch! I forgot `3.6.1` was the correct version 😅

**Comment by vinamartin** [comment]
```
12:51:46 The specified artifact is larger than the set maximum artifact size. 
12:51:46 
12:51:46 	Artifact: /jenkins/workspace/DDF-Jobs/pr/Linux/platform/osgi/platform-osgi-configadmin/target/platform-osgi-configadmin-2.14.0-SNAPSHOT.jar
12:51:46 	Artifact Size: 2472575 Bytes
12:51:46 	Max Artifact Size: 1048576 Bytes
12:51:46 . Either reduce the artifact size (highly recommended) or set the maxArtifactSize property in the enforcer-plugin to a higher value.
```

**Comment by bakejeyner** [comment]
build now
(changed max artifact size for configadmin)

**Comment by bakejeyner** [comment]
build now
(changed max artifact size for configadmin)

**Comment by bakejeyner** [comment]
Looks like CI reported incorrectly:
```
01:37:25 Tests run: 245, Failures: 0, Errors: 0, Skipped: 10, Time elapsed: 2,902.048 sec - in ddf.test.itests.ContainerPerSuiteItestSuite
```
and
```
01:39:45 [INFO] ------------------------------------------------------------------------
01:39:45 [INFO] BUILD SUCCESS
01:39:45 [INFO] ------------------------------------------------------------------------
01:39:45 [INFO] Total time: 02:43 h
01:39:45 [INFO] Finished at: 2019-02-26T01:39:45+00:00
01:39:49 [INFO] Final Memory: 908M/2211M
01:39:49 [INFO] ------------------------------------------------------------------------
01:39:51 Set build name.
01:39:52 New build name is 'PR 4431'
01:39:52 Taking single-use slave mesos-jenkins-860856db708d4ebb9ebf9e262616f83d-linux-large offline.
01:39:52 channel stopped
01:39:54 Archiving artifacts
01:39:54 WARN: No artifacts found that match the file pattern "distribution/test/itests/test-itests-ddf/target/exam/**/*.log, ui/yarn-error.log". Configuration error?
01:39:57 WARN: java.lang.InterruptedException: no matches found within 10000
01:39:59 [BFA] Scanning build for known causes...
01:40:21 [BFA] No failure causes found
01:40:21 [BFA] Done. 21s
01:40:22 Adding one-line test results to commit status...
01:40:22 Setting status of 217ec53c7429973c742d6acea65a8f53a54f671d to FAILURE with url https://jenkins.codice.org/job/DDF-Jobs/job/pr/job/Linux/6288/ and message: 'JOB FAILURE: https://jenkins.codice.org/job/DDF-Jobs/job/pr/job/Linux/6288/
01:40:22  9 tests run, 0 skipped, 0 failed.'
01:40:22 Using context: Internal CI Pipeline
```

**Comment by bakejeyner** [comment]
build now
(want to re-run, since last build may have reported incorrectly)

**Comment by bakejeyner** [comment]
build now
(CI failed on a `wget`. I was able to build the failing module locally.)

**Comment by bakejeyner** [comment]
build now

**Comment by tyler30clemens** [comment]
Hero Successful ✅
verified testing steps mentioned in port worked, including the logout

**Comment by vinamartin** [comment]
Overriding Codacy for this PR since it was a cherry-pick

**Pull request merged**